varinfo/cf_config.py

```diff
@@ -160,40 +155,43 @@ def _create_attributes_object(cf_item: dict) -> dict[str, str]:
             for attribute in cf_item.get('Attributes', {})
         }
 
-    def get_cf_attributes(
-        self, variable: str | None = None
-    ) -> dict[str, dict[str, str]]:
-        """Return the CF overrides and supplements that match a given
-        variable. If a variable is not specified, then return all overrides
-        and supplements. If there are no overrides or supplements, then
-        empty dictionaries will be returned instead.
```

Comment:
Okay, so there are two big changes to this method...

1) The variable path is now a required argument. That means that the output of this method will always be the applicable CF overrides for the named variable. Previously, if the variable path was not specified, all CF overrides for the `CFConfig` object were returned. This made the output inconsistent:

For a variable specific call:

```
{
    'attribute_name': 'attribute_value',
    'other_attribute_name': 'other_attribute_value',
}
```

For an unspecified variable call:

```
{
    '/variable/pattern': {
        'attribute_name': 'attribute_value',
        'other_attribute_name': 'other_attribute_value',
    },
    '/other/variable/pattern': {
        'attribute_name': 'attribute_value',
        'other_attribute_name': 'other_attribute_value',
    },
}
```

Inconsistent output is tricky to code around, and the full dictionary of all CF overrides is available just from the `CFConfig.cf_overrides` class attribute instead.